src/passes/RemoveUnusedModuleElements.cpp

```diff
@@ -214,6 +214,14 @@ struct RemoveUnusedModuleElements : public Pass {
       return analyzer.reachable.count(
                ModuleElement(ModuleElementKind::Event, curr->name)) == 0;
     });
+
+    for (auto& table : module->tables) {
+      table->segments.erase(
+        std::remove_if(table->segments.begin(),
+                       table->segments.end(),
+                       [&](auto& seg) { return seg.data.empty(); }),
```

Comment:
It looks like this was already done below on line 225.

Reply:
No, this removes empty segments first. Consider the case of a table with one empty segment. If we don't remove that segment first, the table would not be removed on 225. I encountered this when I was adding tests.

Reply:
Got it, thanks 👍